### PR TITLE
Remove monolinker build support

### PR DESCRIPTION
**Pull request by marek-safar**
Any unlikely future fixes for it will be done in existing branches

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -17,10 +13,8 @@
     <RunApiCompat>false</RunApiCompat>
   </PropertyGroup>
 
-  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" Condition="'$(MonoBuild)' == ''" />
-  <!-- Still use global versions even if not using arcade. -->
-  <Import Project="eng/Versions.props" Condition="'$(MonoBuild)' != ''" />
-  <Import Project="eng/Analyzers.props" />
+  <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+    <Import Project="eng/Analyzers.props" />
```
Comment (akoeplinger) [inline Directory.Build.props:17]: nit: indentation
```suggestion
  <Import Project="eng/Analyzers.props" />
```

File: README.md [hunk 2]
```diff
@@ -1,29 +1,47 @@
-# IL Linker
+# Trimming Tools
 
-The [IL Linker](src/linker/README.md) is a tool one can use to only ship the minimal possible IL code and metadata that a set of 
-programs might require to run as opposed to the full libraries.
+This repository hosts various tools and msbuild tasks which are used when trimming managed applications with the .NET5 and newer.
```
Comment (akoeplinger) [inline README.md:3]:
```suggestion
This repository hosts various tools and msbuild tasks which are used when trimming managed applications with .NET 5 and newer.
```

File: README.md [hunk 3]
```diff
@@ -1,29 +1,47 @@
...
 
-It is used by the various Xamarin products to extract only the bits of code that are needed to run
-an application on Android, iOS and other platforms.
+## IL Linker
 
-It can also be used in the form of [ILLink.Tasks](src/ILLink.Tasks/README.md) to reduce the size of .NET Core apps.
+The [IL Linker](src/linker/README.md) is the developer's tool than can be used to produce apps that contain only code and assembly dependencies which are necessary to run the app. It's fully integrated into
```
Comment (akoeplinger) [inline README.md:7]:
```suggestion
The [IL Linker](src/linker/README.md) is the developer's tool that can be used to produce apps that contain only code and assembly dependencies which are necessary to run the app. It's fully integrated into
```

File: README.md [hunk 4]
```diff
@@ -1,29 +1,47 @@
...
+.NET SDKs via [ILLink.Tasks](src/ILLink.Tasks/README.md) build task. and exposed via `dotnet publish` trimming [settings](https://docs.microsoft.com/en-us/dotnet/core/deploying/trim-self-contained#trim-your-app---cli).
```
Comment (akoeplinger) [inline README.md:8]:
```suggestion
.NET SDKs via [ILLink.Tasks](src/ILLink.Tasks/README.md) build task and exposed via `dotnet publish` trimming [settings](https://docs.microsoft.com/en-us/dotnet/core/deploying/trim-self-contained#trim-your-app---cli).
```

File: README.md [hunk 5]
```diff
@@ -1,29 +1,47 @@
...
 
-# Analyzer
+The linker is always enabled for all size sensitive .NET workloads like Blazor WebAssembly, Xamarin or .NET mobile and can manually enabled for other project types. The default apps trimming setting can be further customized by using range set of [msbuild properties](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options).
```
Comment (akoeplinger) [inline README.md:10]:
```suggestion
The linker is always enabled for all size sensitive .NET workloads like Blazor WebAssembly, Xamarin or .NET mobile and can be manually enabled for other project types. The default apps trimming setting can be further customized by using a number of [msbuild properties](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options).
```

File: README.md [hunk 6]
```diff
@@ -1,29 +1,47 @@
...
+The [analyzer](src/analyzer/README.md) is a tool to analyze dependencies which were recorded during linker processing. It tracks details about reasons and connection between elements to keep it in the resulting linked assembly. It can be used to better understand the dependencies between different types and members to help further reduce the linked output.
 
-## How to build the IL Linker
+## Trimming Lens
+
+The [tlens](src/tlens/README.md) is another tool for developers which can be used to explore ways to reduce the size of trimmed apps or exploring libraries readiness for trimming. The tool produces a recommendation where the compiled source could be improved to produce even smaller outputs when trimmed using the linker.
+
+## Source Code Analyzer
+
+Another tool available for developers is implemented as [Roslyn Analyzer](src/ILLink.RoslynAnalyzer) which runs on source code and warns developers about code patterns and APIs which are problematic when building code which should could be used with trimmed apps.
```
Comment (akoeplinger) [inline README.md:22]:
```suggestion
Another tool available for developers is implemented as [Roslyn Analyzer](src/ILLink.RoslynAnalyzer) which runs on source code and warns developers about code patterns and APIs which are problematic when building code which could be used with trimmed apps.
```

File: README.md [hunk 7]
```diff
@@ -1,29 +1,47 @@
...
+
+# Contributing
+
+We welcome contributions! Many developers have helped make this project better by reporting [issues](issues) or contributing [pull requests](linker/pulls).
```
Comment (akoeplinger) [inline README.md:26]:
```suggestion
We welcome contributions! Many developers have helped make this project better by reporting [issues](https://github.com/mono/linker/issues) or contributing [pull requests](https://github.com/mono/linker/pulls).
```

File: src/analyzer/analyzer.csproj [hunk 8]
```diff
@@ -7,15 +7,10 @@
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(MonoBuild)' == ''">
-    <DefineConstants>$(DefineConstants);FEATURE_ILLINK</DefineConstants>
+  <PropertyGroup>
     <TargetFramework>net5.0</TargetFramework>
```
Comment (akoeplinger) [inline src/analyzer/analyzer.csproj:11]: nit: combine this with the previous PropertyGroup

File: test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs [hunk 9]
```diff
@@ -194,28 +194,12 @@ internal static void GetDirectoryPaths (out string rootSourceDirectory, out stri
...
-			var tfm = "net5.0";
-#elif NET471
-			var tfm = "net471";
-#else
-			var tfm = "";
-#endif
-
-#if ILLINK
-			// Deterministic builds sanitize source paths, so CallerFilePathAttribute gives an incorrect path.
-			// Instead, get the testcase dll based on the working directory of the test runner.
```
Comment (akoeplinger) [inline test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs:208]: I think this comment should be preserved, or remove the `thisFile` parameter from the method

File: test/Mono.Linker.Tests.Cases/LinkAttributes/LinkerAttributeRemoval.cs [hunk 10]
```diff
@@ -15,13 +15,11 @@ namespace Mono.Linker.Tests.Cases.LinkAttributes
...
 
-#if !ILLINK
-	[Reference ("System.dll")]
-#endif
 	[SetupCompileBefore ("attribute.dll", new[] { "Dependencies/LinkerAttributeRemovalAttributeToRemove.cs" })]
 	[SetupCompileBefore ("copyattribute.dll", new[] { "Dependencies/LinkerAttributeRemovalAttributeFromCopyAssembly.cs" })]
 	[SetupLinkerAction ("copy", "copyattribute")]
-#if !ILLINK
+#if !NETCOREAPP
+	[Reference ("System.dll")]
```
Comment (akoeplinger) [inline test/Mono.Linker.Tests.Cases/LinkAttributes/LinkerAttributeRemoval.cs:22]: is there still a case where this matters if we remove the mono build? illinker isn't used on Windows .NET Framework builds right?

Comment (marek-safar) [reply test/Mono.Linker.Tests.Cases/LinkAttributes/LinkerAttributeRemoval.cs:22]: it's there only to make tests sharing easier with Unity

File: src/analyzer/analyzer.csproj (hunk 8 above)
Comment (marek-safar) [reply src/analyzer/analyzer.csproj:11]: I tried not to change all csproj much, they all deserve another cleanup

File: test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs (hunk 9 above)
Comment (marek-safar) [reply test/ILLink.RoslynAnalyzer.Tests/TestCaseUtils.cs:208]: Fixed

**Review by akoeplinger — APPROVED** [review]

**Pull request merged**